### PR TITLE
[closes #11625 and closes #2488]add global and threadvar(with `--threads:off` mode ) pragmas supports for JS backend

### LINKED ISSUE #2488: [Codegen] {.global.} does not work.
var marked {.global.} will effectively be local in JS.

``` nim
proc test() =
    var i {.global.} = 0
    echo i
    inc i
test()
```

``` js
function test_15003() {
var i_15005 = [0];
rawEcho(cstrToNimstr((i_15005[0])+""));
i_15005[0] = addInt(i_15005[0], 1);
}
```

I guess it's low priority but still worth mentioning.

### LINKED ISSUE #11625: Compiler doesn't warn when using {.global.} and {.threadvar.} on JS backend despite it not working
This includes the `once` template from `system` which depends on the pragma working.

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

**Comment by ringabout** [comment]
Fixed:
#11625
#2488

**Comment by ghost** [comment]
Does threadvar actually work as expected? What if you use it from two different web workers?

**Comment by ringabout** [comment]
What do you mean? JS backend doesn't support multiple threads.

> {.threadvar.} work for nim js, with no warning or error, and have same semantic as nim c --threads:off ; that's the least surprising, most useful behavior, avoiding to special case for nim js.

**Comment by ghost** [comment]
> What do you mean? JS backend doesn't support multiple threads.
> 
> > {.threadvar.} work for nim js, with no warning or error, and have same semantic as nim c --threads:off ; that's the least surprising, most useful behavior, avoiding to special case for nim js.

I meant https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers

**Comment by ringabout** [comment]
Yea, it doesn't work with web worker.
I will edit titles.

File: compiler/jsgen.nim [hunk 1]
```diff
@@ -1736,13 +1736,18 @@ proc genVarInit(p: PProc, v: PSym, n: PNode) =
     varCode: string
     varName = mangleName(p.module, v)
     useReloadingGuard = sfGlobal in v.flags and p.config.hcrOn
+    useGlobalPragmas = sfGlobal in v.flags and (sfPure in v.flags or sfThread in v.flags)
```
Comment (Araq) [inline compiler/jsgen.nim]:
```suggestion
    useGlobalPragmas = sfGlobal in v.flags and ({sfPure, sfThread} * v.flags != {})
```

**Comment by timotheecour** [comment]
@Yardanico support for webworkers can be done as future work when `--threads:on` (or some js specific flag) is specified, but this PR is correct in absence of `--threads:on`

**Pull request merged**